ext/ldap/ldap.stub.php

```diff
@@ -4,281 +4,281 @@
 
 #ifdef HAVE_ORALDAP
 /** @return resource|false */
-function ldap_connect(string $hostname = UNKNOWN, int $port = 389, string $wallet = UNKNOWN, string $wallet_passwd = UNKNOWN, int $authmode = GSLC_SSL_NO_AUTH) {}
+function ldap_connect(?string $uri = null, int $port = 389, string $wallet = UNKNOWN, string $password = UNKNOWN, int $auth_mode = GSLC_SSL_NO_AUTH) {}
 #else
 /** @return resource|false */
-function ldap_connect(string $hostname = UNKNOWN, int $port = 389) {}
+function ldap_connect(?string $uri = null, int $port = 389) {}
 #endif
 
-/** @param resource $link_identifier */
-function ldap_unbind($link_identifier): bool {}
+/** @param resource $ldap */
+function ldap_unbind($ldap): bool {}
 
 /**
- * @param resource $link_identifier
+ * @param resource $ldap
  * @alias ldap_unbind
  */
-function ldap_close($link_identifier): bool {}
+function ldap_close($ldap): bool {}
 
-/** @param resource $link_identifier */
-function ldap_bind($link_identifier, string $bind_rdn = UNKNOWN, string $bind_password = UNKNOWN): bool {}
+/** @param resource $ldap */
+function ldap_bind($ldap, ?string $dn = null, ?string $password = null): bool {}
 
 /**
- * @param resource $link_identifier
+ * @param resource $ldap
  * @return resource|false
  */
-function ldap_bind_ext($link_identifier, string $bind_rdn = UNKNOWN, string $bind_password = UNKNOWN, array $servercontrols = []) {}
+function ldap_bind_ext($ldap, ?string $dn = null, ?string $password = null, array $controls = []) {}
 
 #ifdef HAVE_LDAP_SASL
-/** @param resource $link */
-function ldap_sasl_bind($link, string $binddn = UNKNOWN, string $password = UNKNOWN, string $sasl_mech = UNKNOWN, string $sasl_realm = UNKNOWN, string $sasl_authc_id = UNKNOWN, string $sasl_authz_id = UNKNOWN, string $props = UNKNOWN): bool {}
+/** @param resource $ldap */
+function ldap_sasl_bind($ldap, ?string $dn = null, ?string $password = null, ?string $mech = null, ?string $realm = null, ?string $authc_id = null, ?string $authz_id = null, ?string $props = null): bool {}
 #endif
 
 /**
- * @param resource|array $link_identifier
+ * @param resource|array $ldap
  * @return resource|false
  */
-function ldap_read($link_identifier, array|string $base_dn, array|string $filter, array $attributes = [], int $attrsonly = 0, int $sizelimit = -1, int $timelimit = -1, int $deref = LDAP_DEREF_NEVER, array $servercontrols = []) {}
+function ldap_read($ldap, array|string $base, array|string $filter, array $attributes = [], int $attrsonly = 0, int $sizelimit = -1, int $timelimit = -1, int $deref = LDAP_DEREF_NEVER, array $controls = []) {}
```

Comment:
```suggestion
function ldap_read($ldap, array|string $base, array|string $filter, array $attributes = [], int $attributes_only = 0, int $size_limit = -1, int $time_limit = -1, int $deref = LDAP_DEREF_NEVER, array $controls = []) {}
```

Reply:
See above, @nikic proposed the same, but both openldap and php use sizelimit and timelimit in other places, it’s always have been a whole word.

I can change $attrsonly though.

Reply:
I don't see a big problem with diverging from these other names, but I also don't mind if these stay without the `_`. So I'm ok.